### PR TITLE
One file per datareader

### PR DESCRIPTION
**Pull request by femtotrader**
Should fix https://github.com/pydata/pandas-datareader/issues/58

### CODE REVIEW

**Comment by femtotrader** [comment]
@davidastephens @bashtage Could you have a look ?

File: pandas_datareader/data.py [hunk 1]
```diff
@@ -84,1225 +69,11 @@ def DataReader(name, data_source=None, start=None, end=None,
                                       retry_count=retry_count, pause=pause)
     elif data_source == "google":
         return get_data_google(symbols=name, start=start, end=end,
-                               adjust_price=False, chunksize=25,
-                               retry_count=retry_count, pause=pause)
+                               chunksize=25, retry_count=retry_count, pause=pause)
```
Comment (davidastephens) [inline pandas_datareader/data.py]: Why is adjust_price removed?

**Comment by davidastephens** [comment]
Do we need the datareaders package? Can it just be pandas_datareader/google/..?

**Comment by femtotrader** [comment]
I created the `datareaders` package because I didn't like the fact of having for example `fred.py` `famafrench.py`, `google` and `yahoo` in the same directory than `date_chunks.py`, `shared.py` and `url_request.py`

I can remove it if you think that's more understandable and put `date_chunks.py`, `shared.py` and `url_request.py` in a `commons` package.

`adjust_price` is removed because Google don't provide it (only Yahoo)
That's the reason why 

    if ret_index:
        hist_data['Ret_Index'] = _calc_return_index(hist_data['Adj Close'])
    if adjust_price:
        hist_data = _adjust_prices(hist_data)

is in `get_data_yahoo`

**Comment by femtotrader** [comment]
@davidastephens I have update accordingly.

**Comment by femtotrader** [comment]
I also think that some cleanup is / will be necessary.

For example, 

- in `famafrench.py`
  - `_FAMAFRENCH_URL` should be named `_URL`
  - `get_data_famafrench` should be name only `get_data`

most files should have such cleanup but maybe we should do it after merging.

`options.py` should also be improved because it's in `yahoo` so there is no need for `data_source``

we should have an other `options.py` at same level than `data.py`

**Comment by femtotrader** [comment]
Options is now also a function in `data.py` which is able to dispatch per data vendor. For now there is only one data "vendor" for options : Yahoo.

**Comment by davidastephens** [comment]
I think for now, given there is only 2 things in the common package, we can just have it as common.py in the base package.

**Comment by davidastephens** [comment]
I think we should merge the package structure changes and do the clean-up and option dispach in 2 separate pulls.  

Can you rebase against master and squash to one commit (except the option dispatch and clean-up?).

Thanks for all the work on this.

**Comment by femtotrader** [comment]
Hum... that will be quite difficult for me. I'm not very good with git. Maybe I should create an other branch ?

**Comment by davidastephens** [comment]
There is a good tutorial here:
https://github.com/pydata/pandas/wiki/Using-Git

Here are the steps you need to do:
```
#add the pandas remote repository to your fork and name it upstream

git remote add upstream git://github.com/pydata/pandas-datareader

#fetch and then rebase interactively to squash, reword, fix, otherwise change some commits

git fetch upstream
git rebase --interactive upstream/master

```

In the text file you get from rebasing, just remove the Option and cleanup commits (delete the lines) and change the rest of them from pick to squash (except the first one).  That squashes them all into the first commit.

**Comment by femtotrader** [comment]
Closing this PR is favor of https://github.com/pydata/pandas-datareader/pull/68

**Comment by femtotrader** [comment]
Sorry

**Pull request closed without merging**